### PR TITLE
Improvements to the DXR Wizard

### PR DESCRIPTION
**Pull request by anisunity**
On user request, we decided to add new checks and additional informaiton in the DXR wizard. Now the DXR wizard displays a warning message for unchecked feature in the HDRP asset and and an info message for the corresponding an unchecked default frame setting. The wizard also displays the name of the target asset that may require modification.

image

Testing status
Clicked on all the asset features and default camera frame settings and made sure the right messages popped when they are checked/unchecked.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs [hunk 1]
```diff
@@ -137,18 +138,30 @@ public ConfigStyle(string label, string error, string button = resolve, MessageT
                 label: "Direct3D 12",
                 error: "Direct3D 12 is needed! (Editor restart is required)");
             public static readonly ConfigStyle dxrScreenSpaceShadow = new ConfigStyle(
-                label: "Screen Space Shadow",
-                error: "Screen Space Shadows are disabled in the current HDRP asset. You will not be able to toggle ray traced shadows on the lights in your scene. You can enable the feature in the HDRP asset under Lighting -> Shadows -> Screen Space Shadows", messageType: MessageType.Info);
+                label: "Screen Space Shadows (Asset)",
+                error: "Screen Space Shadows are disabled in the current HDRP asset. You will not be able to toggle ray traced shadows on the lights in your scene. You can enable the feature in the HDRP asset under Lighting > Shadows > Screen Space Shadows", messageType: MessageType.Warning);
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs:142]: Screen Space Shadows are disabled in the current HDRP Asset which means you cannot enable ray-traced shadows for lights in your scene. To enable this feature, open your HDRP Asset, go to **Lighting > Shadows**, and enable **Screen Space Shadows**.

File: com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs [hunk 2]
```diff
@@ -137,18 +138,30 @@ public ConfigStyle(string label, string error, string button = resolve, MessageT
...
+                label: "Screen Space Shadows (Default Camera Frame Setting)",
+                error: "Screen Space Shadows are disabled in the default Camera Frame setting. You can enable the feature in Project Settings > HDRP Default Settings > Frame Settings > Default Frame Settings For Camera > Lighting > Screen Space Shadows", messageType: MessageType.Info);
             public static readonly ConfigStyle dxrReflections = new ConfigStyle(
-                label: "Reflections",
-                error: "Screen Space Reflections are disabled in the current HDRP asset. You will not be able to toggle ray traced reflections though your volume components. You can enable the feature in the HDRP asset under Lighting -> Reflections -> Screen Space Reflections", messageType: MessageType.Info);
+                label: "Reflection (Asset)",
+                error: "Screen Space Reflection is disabled in the current HDRP asset. You will not be able to toggle ray traced reflections though your volume components. You can enable the feature in the HDRP asset under Lighting > Reflections > Screen Space Reflections", messageType: MessageType.Warning);
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs:148]: Screen Space Reflection is disabled in the current HDRP Asset which means you cannot enable ray-traced reflections in Volume components. To enable this feature, open your HDRP Asset, go to **Lighting > Reflections**, and enable **Screen Space Reflections**.

File: com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs [hunk 3]
```diff
@@ -137,18 +138,30 @@ public ConfigStyle(string label, string error, string button = resolve, MessageT
...
             public static readonly ConfigStyle dxrTransparentReflections = new ConfigStyle(
-                label: "Transparent Reflections",
-                error: "Transparent Screen Space Reflections are disabled in the current HDRP asset. You will not be able to toggle ray traced reflections on transparent objects though your volume components. You can enable the feature in the HDRP asset under Lighting -> Reflections -> Transparent Screen Space Reflections", messageType: MessageType.Info);
+                label: "Transparent Reflection (Asset)",
+                error: "Transparent Screen Space Reflection is disabled in the current HDRP asset. You will not be able to toggle ray traced reflections on transparent objects though your volume components. You can enable the feature in the HDRP asset under Lighting > Reflections > Transparent Screen Space Reflections", messageType: MessageType.Warning);
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs:154]: Transparent Screen Space Reflection is disabled in the current HDRP Asset which means you cannot enable ray-traced reflections for transparent GameObjects from Volume components. To enable this feature, open your HDRP Asset, go to **Lighting > Reflections**, and enable **Transparent Screen Space Reflections**.

File: com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs [hunk 4]
```diff
@@ -137,18 +138,30 @@ public ConfigStyle(string label, string error, string button = resolve, MessageT
...
+                error: "Transparent Screen Space Reflection is disabled in the default Camera Frame setting. You can enable the feature in Project Settings > HDRP Default Settings > Frame Settings > Default Frame Settings For Camera > Lighting > On Transparent", messageType: MessageType.Info);
             public static readonly ConfigStyle dxrGI = new ConfigStyle(
-                label: "Global Illumination",
-                error: "Screen Space Global Illumination is disabled in the current HDRP asset. You will not be able to toggle ray global illumination though your volume components. You can enable the feature in the HDRP asset under Lighting -> Screen Space Global Illumination", messageType: MessageType.Info);
-			public static readonly ConfigStyle dxr64bits = new ConfigStyle(
+                label: "Global Illumination (Asset)",
+                error: "Screen Space Global Illumination is disabled in the current HDRP asset. You will not be able to toggle ray global illumination though your volume components. You can enable the feature in the HDRP asset under Lighting -> Screen Space Global Illumination", messageType: MessageType.Warning);
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs:160]: **Screen Space Global Illumination** is disabled in the current HDRP asset which means you cannot enable ray-traced global illumination in Volume components. To enable this feature, open your HDRP Asset, go to **Lighting** and enable **Screen Space Global Illumination**.

File: com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs [hunk 5]
```diff
@@ -137,18 +138,30 @@ public ConfigStyle(string label, string error, string button = resolve, MessageT
...
-                label: "Screen Space Shadow",
-                error: "Screen Space Shadows are disabled in the current HDRP asset. You will not be able to toggle ray traced shadows on the lights in your scene. You can enable the feature in the HDRP asset under Lighting -> Shadows -> Screen Space Shadows", messageType: MessageType.Info);
+                label: "Screen Space Shadows (Asset)",
+                error: "Screen Space Shadows are disabled in the current HDRP asset. You will not be able to toggle ray traced shadows on the lights in your scene. You can enable the feature in the HDRP asset under Lighting > Shadows > Screen Space Shadows", messageType: MessageType.Warning);
+            public static readonly ConfigStyle dxrScreenSpaceShadowFS = new ConfigStyle(
+                label: "Screen Space Shadows (Default Camera Frame Setting)",
+                error: "Screen Space Shadows are disabled in the default Camera Frame setting. You can enable the feature in Project Settings > HDRP Default Settings > Frame Settings > Default Frame Settings For Camera > Lighting > Screen Space Shadows", messageType: MessageType.Info);
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs:145]: Screen Space Shadows are disabled in the default Camera Frame Settings. This means Cameras that use these Frame Settings do not render ray-traced shadows. To enable this feature, go to **Project Settings > HDRP Default Settings > Frame Settings > Default Frame Settings For Camera > Lighting** and enable **Screen Space Shadows**.

File: com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs [hunk 6]
```diff
@@ -137,18 +138,30 @@ public ConfigStyle(string label, string error, string button = resolve, MessageT
...
-                label: "Reflections",
-                error: "Screen Space Reflections are disabled in the current HDRP asset. You will not be able to toggle ray traced reflections though your volume components. You can enable the feature in the HDRP asset under Lighting -> Reflections -> Screen Space Reflections", messageType: MessageType.Info);
+                label: "Reflection (Asset)",
+                error: "Screen Space Reflection is disabled in the current HDRP asset. You will not be able to toggle ray traced reflections though your volume components. You can enable the feature in the HDRP asset under Lighting > Reflections > Screen Space Reflections", messageType: MessageType.Warning);
+            public static readonly ConfigStyle dxrReflectionsFS = new ConfigStyle(
+                label: "Reflection (Default Camera Frame Setting)",
+                error: "Screen Space Reflection is disabled in the default Camera Frame setting. You can enable the feature in Project Settings > HDRP Default Settings > Frame Settings > Default Frame Settings For Camera > Lighting > Screen Space Reflections", messageType: MessageType.Info);
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs:151]: Screen Space Reflection is disabled in the default Camera Frame Settings. This means Cameras that use these Frame Settings do not render ray-traced reflections. To enable this feature, go to **Project Settings > HDRP Default Settings > Frame Settings > Default Frame Settings For Camera > Lighting** and enable **Screen Space Reflections**.

File: com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs [hunk 7]
```diff
@@ -137,18 +138,30 @@ public ConfigStyle(string label, string error, string button = resolve, MessageT
...
+                label: "Transparent Reflection (Asset)",
+                error: "Transparent Screen Space Reflection is disabled in the current HDRP asset. You will not be able to toggle ray traced reflections on transparent objects though your volume components. You can enable the feature in the HDRP asset under Lighting > Reflections > Transparent Screen Space Reflections", messageType: MessageType.Warning);
+            public static readonly ConfigStyle dxrTransparentReflectionsFS = new ConfigStyle(
+                label: "Transparent Reflection (Default Camera Frame Setting)",
+                error: "Transparent Screen Space Reflection is disabled in the default Camera Frame setting. You can enable the feature in Project Settings > HDRP Default Settings > Frame Settings > Default Frame Settings For Camera > Lighting > On Transparent", messageType: MessageType.Info);
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs:157]: Transparent Screen Space Reflection is disabled in the default Camera Frame Settings. This means Cameras that use these Frame Settings do not render ray-traced reflections for transparent GameObjects. To enable this feature, go to Project Settings > HDRP Default Settings > Frame Settings > Default Frame Settings For Camera > Lighting and enable On Transparent.

File: com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs [hunk 8]
```diff
@@ -137,18 +138,30 @@ public ConfigStyle(string label, string error, string button = resolve, MessageT
...
-			public static readonly ConfigStyle dxr64bits = new ConfigStyle(
+                label: "Global Illumination (Asset)",
+                error: "Screen Space Global Illumination is disabled in the current HDRP asset. You will not be able to toggle ray global illumination though your volume components. You can enable the feature in the HDRP asset under Lighting -> Screen Space Global Illumination", messageType: MessageType.Warning);
+            public static readonly ConfigStyle dxrGIFS = new ConfigStyle(
+                label: "Global Illumination (Default Camera Frame Setting)",
+                error: "Screen Space Global Illumination is disabledin the default Camera Frame setting. You can enable the feature in Project Settings > HDRP Default Settings > Frame Settings > Default Frame Settings For Camera > Lighting > Screen Space Global Illumination", messageType: MessageType.Info);
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Editor/Wizard/HDWizard.Window.cs:163]: Screen Space Global Illumination is disabled in the default Camera Frame Settings. This means Cameras that use these Frame Settings do not render ray-traced global illumination. To enable this feature, go to **Project Settings > HDRP Default Settings > Frame Settings > Default Frame Settings For Camera > Lighting** and enable **Screen Space Global Illumination**.

**Review by JordanL8 — CHANGES_REQUESTED** [review]
Reworded a few bits.

**Comment by anisunity** [comment]
corrections done

**Review by RSlysz — APPROVED** [review]
lgtm

**Review by JordanL8 — APPROVED** [review]

**Pull request merged**